cmd/api.go

```diff
@@ -161,6 +161,9 @@ After the scaffold is written, api will run make on the project.
 	make run
 `,
 		Run: func(cmd *cobra.Command, args []string) {
+			cmd.MarkFlagRequired("group")
```

Comment:
I have tried locally, the way you wrote it doesn't seem to work.
I tweak it a bit. I found the following way works. Please give it a try.
It can give an error like `Error: required flag(s) "kind" not set`.
```
func newAPICommand() *cobra.Command {
	options := apiOptions{
		apiScaffolder: scaffold.API{},
	}

	apiCmd := &cobra.Command{
		Use:   "api",
		Short: "Scaffold a Kubernetes API",
		Long: `blah`,
		Example: `blah`,
		Run: func(cmd *cobra.Command, args []string) {
			options.runAddAPI()
		},
	}

	options.bindCmdFlags(apiCmd)
	apiCmd.MarkFlagRequired("group")
	apiCmd.MarkFlagRequired("version")
	apiCmd.MarkFlagRequired("kind")

	return apiCmd
}
```

Reply:
Hi @mengqiy,
 
It will just work if we pass the 3 args as follows. 

```go
 $ kubebuilder create api --group kubebuilder --version v1alpha1 --kind Foobar
Create Resource [y/n]
y
Create Controller [y/n]
y
Writing scaffold for you to edit...
api/v1alpha1/foobar_types.go
controllers/foobar_controller.go
Running make...
/Users/camilamacedo/go/bin/controller-gen object:headerFile=./hack/boilerplate.go.txt paths="./..."
go fmt ./...
go vet ./...
go build -o bin/manager main.go
camilamacedo@Camilas-MacBook-Pro ~/go/src/example $ 
```

```go
See that the error is just faced if we do not inform value for the Kind. 

 ~/go/src/example $ kubebuilder create api --group kubebuilder --version v1alpha1 --kind 
Error: flag needs an argument: --kind
Usage:
```

Reply:
How you face the error? Are you passing the Kind?

Reply:
To fix https://github.com/kubernetes-sigs/kubebuilder/issues/1153, I believe you need to mark flag as required outside of `Run` method.
 The code probably should look like what I proposed in https://github.com/kubernetes-sigs/kubebuilder/pull/1163/files#r344915843

Reply:
Hi @mengqiy, 

Your propose is not solving the issue and keeping the previous behaviour. 
See:
<img width="982" alt="Screenshot 2019-11-13 at 01 34 28" src="https://user-images.githubusercontent.com/7708031/68755927-327ce000-0601-11ea-8cd3-e42d05824bb4.png">

If we do as you suggested then, the tag `--version` will be set as the value of `--group` which is exactly the issue hat this PR solves. 

NOTE: Also, I could reproduce the scenario where it still not solved here `kubebuilder create api --group --version x --kind D` too.

So, I solved it properly in another way. Let's me know what do you think?